Twispay WooCommerce Plugin/twispay/views/payment-confirmation.php

```diff
@@ -1,297 +1,253 @@
-    }
-    
-    if ( strpos( $encrypted, ',' ) !== false ) {
-        $encryptedParts = explode( ',', $encrypted, 2 );
-        $iv = base64_decode( $encryptedParts[0] );
-        if ( false === $iv ) {
-            return false;
-        }
-        
-        $encrypted = base64_decode( $encryptedParts[1] );
-        if ( false === $encrypted ) {
-            return false;
-        }
-        
-        $decrypted = openssl_decrypt( $encrypted, 'aes-256-cbc', $apiKey, OPENSSL_RAW_DATA, $iv );
-        if ( false === $decrypted ) {
-            return false;
-        }
-        
-        return $decrypted;
-    }
-    
-    return null;
+
+/* Load languages */
+$lang = explode( '-', get_bloginfo( 'language' ) )[0];
+if ( file_exists( TWISPAY_PLUGIN_DIR . 'lang/' . $lang . '/lang.php' ) ){
+    require( TWISPAY_PLUGIN_DIR . 'lang/' . $lang . '/lang.php' );
+} else {
+    require( TWISPAY_PLUGIN_DIR . 'lang/en/lang.php' );
 }
 
-function twispay_tw_log( $string = false ) {
-    $log_file = dirname( __FILE__ ) . '/../twispay-log.txt';
-    
-    if ( ! $string ) {
-        $string = PHP_EOL . PHP_EOL;
-    }
-    else {
-        $string = "[" . date( 'Y-m-d H:i:s' ) . "] " . $string;
+/* Require the "Twispay_TW_Helper_Response" class. */
+require_once( TWISPAY_PLUGIN_DIR . DIRECTORY_SEPARATOR . 'helpers' . DIRECTORY_SEPARATOR . 'Twispay_TW_Helper_Response.php' );
+/* Require the "Twispay_TW_Default_Thankyou" class. */
+require_once( TWISPAY_PLUGIN_DIR . DIRECTORY_SEPARATOR . 'helpers' . DIRECTORY_SEPARATOR . 'Twispay_TW_Default_Thankyou.php' );
+
+
+/* Validate if 'WooCommerce' is NOT installed. */
+if ( !class_exists('WooCommerce') ){
+    ?>
+        <div class="error notice" style="margin-top: 20px;">
+            <h3><?= $tw_lang['general_error_title']; ?></h3>
+            <p><?= $tw_lang['no_woocommerce_f']; ?> <a target="_blank" href="https://wordpress.org/plugins/woocommerce/"><?= $tw_lang['no_woocommerce_s']; ?></a></p>
+        </div>
+    <?php
+
+    exit();
+}
+
+
+/* Get configuration from database. */
+global $wpdb;
+$configuration = $wpdb->get_row( "SELECT * FROM " . $wpdb->prefix . "twispay_tw_configuration" );
+
+
+$secretKey = '';
+if ( $configuration ) {
+    if ( 1 == $configuration->live_mode ) {
+        $secretKey = $configuration->live_key;
+    } else if ( 0 == $configuration->live_mode ) {
+        $secretKey = $configuration->staging_key;
+    } else {
+        /* TODO: Error? */
     }
-    
-    @file_put_contents( $log_file, $string . PHP_EOL, FILE_APPEND );
 }
 
-function twispay_tw_getResultStatuses() {
-    return array( 'complete-ok' );
+
+/* Check if the POST is corrupted: Doesn't contain the 'opensslResult' and the 'result' fields. */
+                                          /* OR */
+/* Check if the 'backUrl' is corrupted: Doesn't contain the 'secure_key' field. */
+if( ((FALSE == isset($_POST['opensslResult'])) && (FALSE == isset($_POST['result']))) || (FALSE == isset($_GET['secure_key'])) ) {
```

Comment:
what is this secure_key in get? if it is the twispay api key (secret key) this is not good. if it is a hash you generated to validate the return url then everything is ok.

Reply:
It represents that "cart_hash" that is added to the "backUrl" when the order is placed and is extracted from the order data.
It is used to validate that the content of the order received through "externalOrderId" contains the same products.